### DISCUSSION TITLE
async for message in stream, high cpu usages.

### DISCUSSION THREAD

**Discussion by hongfy197** [microsoft/autogen#4772] [Q&A]
```
team.run_stream(task=input.message)
async for message in stream:
   pass
```

`async for message in stream` cause high cpu usages.

<img width="994" alt="Screenshot 2024-12-20 at 5 40 20 in the afternoon" src="https://github.com/user-attachments/assets/d5c310a3-fa0b-429d-be94-8f0d81e64e54" />

**Comment by jackgerrits** [comment]
Thanks, I've created an issue for this #4773 

I think I know the cause of the issue, as described in the issue

**Reply by Ocrosoft** [reply]
When using the in-process runtime with the .NET SDK, it can also result in high CPU usage when there are no messages in the queue.

**Reply by CodeRecluse** [reply]
Hey,
i am still facing this issue i am using this in my fastapi backend it just throttling the CPU to 100% slowing down the server they are telling it is fixed but I don't see any improvement.

did you find any workaround ??
It would be helpful, thanks
